### PR TITLE
Fix local image path when using a custom device

### PR DESCRIPTION
**Pull request by jonastreub**
This fixes a bug where you cannot set device images for custom devices.

### CODE REVIEW

**Comment by koenbok** [comment]
Do I need to review this @jonastreub ?

**Comment by jonastreub** [comment]
Yes please!

File: framer/Components/DeviceComponent.coffee [hunk 1]
```diff
@@ -73,6 +73,8 @@ class exports.DeviceComponent extends BaseClass
 		if @_setupDone
 			return
 
+		@_buildInDevices = _.keys Devices
```
Comment (koenbok) [inline framer/Components/DeviceComponent.coffee]: English: builtInDevices. Also we prefer `_.keys(Devices)`.

File: framer/Components/DeviceComponent.coffee [hunk 2]
```diff
@@ -279,19 +281,24 @@ class exports.DeviceComponent extends BaseClass
...
 			return name
 
-		# If we're running Framer Studio and have local files, we'd like to use those
-		if Utils.isFramerStudio() and window.FramerStudioInfo
-			resourceUrl = window.FramerStudioInfo.deviceImagesUrl
-		
-		# If not, we want to get these image from our public resources server
-		else
-			resourceUrl = "//resources.framerjs.com/static/DeviceResources"
+		if @_deviceType in @_buildInDevices
```
Comment (koenbok) [inline framer/Components/DeviceComponent.coffee]: Why not early exit:

```
if @_deviceType in @_buildInDevices
  return name
```

File: test/tests/DeviceComponentTest.coffee [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+			"screenHeight": 1000
+			"deviceImage": "images/framer-icon.png"
+			"deviceImageWidth": 800
+			"deviceImageHeight": 1203
+
+		# Set custom device
+		comp.deviceType = "test"
+
+		comp.deviceType.should.equal "test"
+		comp._deviceImageUrl(comp._deviceImageName()).should.eql "images/framer-icon.png"
```
Comment (koenbok) [inline test/tests/DeviceComponentTest.coffee]: I'd like to see all permutations tested:

- In Framer Studio
- Not in Framer Studio
- Built in device
- Custom device

So, in total 2 x 2 tests.

File: framer/Components/DeviceComponent.coffee (hunk 1 above)
Comment (koenbok) [inline framer/Components/DeviceComponent.coffee]: Also, in theory it's a waste to set this for every instance. You should make it a static global in the module.

File: framer/Components/DeviceComponent.coffee [hunk 4]
```diff
@@ -278,20 +278,23 @@ class exports.DeviceComponent extends BaseClass
 		# If the image is externally hosted, we'd like to use that
 		if _.startsWith(name, "http://") or _.startsWith(name, "https://")
```
Comment (koenbok) [inline framer/Components/DeviceComponent.coffee]: Refactor this a bit so you end up with if...return if...return if...return.

File: test/tests/DeviceComponentTest.coffee (hunk 3 above)
Comment (koenbok) [inline test/tests/DeviceComponentTest.coffee]: Nice.

**Comment by jonastreub** [comment]
Thanks for the feedback @koenbok! Code is much clearer now.

**Pull request merged**